### PR TITLE
fixed broken keras multi input model in gradient explainer

### PR DESCRIPTION
**Pull request by moritzaugustin**
`DeepExplainer` worked nicely with my multi input keras model, but in order to make the `GradientExplainer` work as well I had to copy the respective lines from `deep/deep_tf.py` to `gradient.py`.

Without the fix I would be receiving the following errors when calling the `shap_values` method of the `GradientExplainer` object:
```
File "shap/explainers/gradient.py", line 196, in shap_values
    assert type(X) != list, "Expected a single tensor model input!"

### CODE REVIEW

**Comment by moritzaugustin-tiplu** [comment]
( having changed via force-pushing only my mail address to a noreply address ;-) )

**Comment by slundberg** [comment]
Thanks! Sorry for the slow reply, I'll work on reviewing this soon.

**Comment by moritzaugustin** [comment]
The last two commits can be ignored if they are not wanted... ( I should have created a separate branch )

**Comment by slundberg** [comment]
Thanks again. Both those last two commits look good to me. So let's keep them. I'll merge this for now, and at some point we need to add a unit test for the multi-input test case.

**Comment by niloofardadras** [comment]
@moritzaugustin  I am trying to use gradient ex plainer but i get the error still

assert type(X) != list, "Expected a single tensor model input!"

is this change supposed to fix the error?

**Comment by moritzaugustin** [comment]
in my case it did but i have not tested the current version with tf2 support yet. this code part seems to have been slightly refactored

**Comment by FarzanT** [comment]
Hi,
I'm using Pytorch 1.7 and I still get this error when I pass a list of tensors at the `e.shap_values` call, and interestingly not at the `shap.DeepExplainer` call.  Would you please have a look?
